buildattr64/buildattr64.rst

```diff
@@ -0,0 +1,904 @@
+have no value. Either way, in practice an attempted combination is
+expected to fail in a way specific to a tool chain’s compatibility
+model (for example by provoking a link-time diagnostic).
+
+Forcing Functions
+^^^^^^^^^^^^^^^^^
+
+In many cases the value of an attribute is independent of the value
+of all other attributes. There can be cases where attribute values
+are linked. For example to describe the signing schema in (PAuthABI64_)
+which is described as a tuple (vendor, version) two attributes are
+required. Incompatibilites also exist, such as in (ADDENDA32_)
+``Tag_Advanced_SIMD_arch`` cannot be set if ``Tag_CPU_arch_profile``
+is set to 'M'.
+
+The specification will note known forcing functions. Implementation of
+forcing functions is Q-o-I.
+
+Representing build attributes in a relocatable ELF Object file
+==============================================================
+
+Encoding
+--------
+
+Encoding build attributes are encoded in a vendor-specific section of
+type ``SHT_AARCH64_ATTRIBUTES`` and name ``.ARM.attributes`` (for
+further details see [AAELF64_]).
+
+An attribute is encoded in a <tag, value> pair.
+
+Both tags and numerical values are encoded using unsigned LEB128
+encoding (ULEB128), DWARF style (for details see [GDWARF_]).
+
+String values are encoded using NUL-terminated byte strings (NTBS).
+
+Encoding of meta data in a loadable-unit
+----------------------------------------
+
+Build attributes are only defined for relocatable object files.  The
+encoding of metadata in a loadable-unit (executable or shared-library)
+is platform specific.  For example a platform may choose to use GNU
+program properties as defined in (LINUX_ABI_). This document specifies
+the meaning of build attributes for relocatable object files only.  The
+presence of ``.ARM.attributes`` sections in a loadable-unit is
+permitted, but the interpretation of the contents of the section is
+outside the scope of this document. The use of ``.note.gnu.property``
+sections for loadable-units is recommended for SysVr4 platforms,
+see (SYSVABI64_) for details.
+
+.. raw:: pdf
+
+   PageBreak
+
+Formal syntax of an ELF Attributes Section
+==========================================
+
+An ELF Attributes section uses a processor specific section with details
+
+.. table:: Attributes Section
+
+    +---------------------+----------------------------+-------+-----------+------+------------+
+    | Name                | Type                       | Flags | Alignment | Info | Entry Size |
+    +=====================+============================+=======+===========+======+============+
+    | ``.ARM.attributes`` | ``SHT_AARCH64_ATTRIBUTES`` | 0x0   | 1         | 0    | 0          |
+    +---------------------+----------------------------+-------+-----------+------+------------+
+
+A consumer may not assume the natural alignment of a data type such as ``uint32``.
+
+The syntactic structure of an attributes section is::
+
+   <format-version: <version-letter>>
```

Comment:
The  <format-version: <version-letter>> (double '<, >' also around 'version-letter') and <uint32: subsection-length> (single '<, >', not around 'subsection-length') does not indicated a different between the cases.